user_guide_src/source/installation/deployment.rst

```diff
@@ -59,9 +59,31 @@ See :ref:`file-locator-caching`.
 PHP Preloading
 ==============
 
-If you want to use `Preloading <https://www.php.net/manual/en/opcache.preloading.php>`_,
-we provide a
-`preload script <https://github.com/codeigniter4/CodeIgniter4/blob/develop/preload.php>`_.
+With PHP preloading, you can instruct the server to load essential files like functions and classes into memory during startup.
+This means these elements are readily available for all requests, skipping the usual loading process and boosting your
+application's performance. However, this comes at the cost of increased memory usage and requires restarting the
+PHP engine for changes to take effect.
+
+.. note:: If you want to use `Preloading <https://www.php.net/manual/en/opcache.preloading.php>`_,
+    we provide a `preload script <https://github.com/codeigniter4/CodeIgniter4/blob/develop/preload.php>`_.
+
+Requirement
+-----------
+
+Preloading for classes that are used more than once in the same server is not possible. You must isolated application to *dedicated* server,
```

Comment:
The 2nd sentence is not clear. Do you mean to 1) "You must have an isolated application..." or 2) "You must isolate the application...". Either way means the same although subtle difference: 1 if you want as a requirement or 2 if you want as command. Please change as you deem okay,

Reply:
Preloading is to load specified files at the web server startup. So if you try to load (require/include) the file that is already loaded, a Fatail error like `Cannot declare class XXX, because the name is already in use` occurs.

If you don't call require/include the same file more than once, it is theoretically possible to run multiple apps. However, if, for example, the two apps have different versions of the framework, only one class/function with the same name can be loaded, so one of the apps will run a different version of the class/function. 

Incidentally, CI3 has functions with the same name, such as `log_message()`, which will not work with the following error:

> Fatal error: Uncaught Error: Undefined constant "CodeIgniter\Cache\FactoriesCache\WRITEPATH" in ... /CodeIgniter4/system/Autoloader/FileLocatorCached.php:56

Also, preloading files for completely unrelated apps is only a disadvantage, so realistically, we would have one app running on one web server.

Reply:
I already test, have more apps in virtualhost Dev server use php8-*

1. Use handler php8-3
2. Use handler php8-2
3. Use handler php8-1
More

Its 1 handler PHP for one apps, my web server just one (Apache). Same as nginx can use more handler php

Reply:
I think just clearing what a mean "isolated" and "dedicated"

Reply:
I haven't actually tried, but, if you have more than one PHP for a web server, you could handle more than one app with preload with one server.

Reply:
@kenjis @paulbalandan Check again, maybe now it's right explanation

Reply:
By the way, what is "handler php8-3" exactly? Can you show sample configuration?

Reply:
@kenjis In Apache2

```bash
# Handler PHP
<FilesMatch \.php$>
        # For Apache
        SetHandler "proxy:unix:/run/php/php8.3-fpm.sock|fcgi://localhost"
</FilesMatch>
```

Reply:
It is configuration for PHP-FPM. But our user guide is based on Apache mod_php:
https://codeigniter4.github.io/CodeIgniter4/installation/running.html#hosting-with-apache

How about this? > @ddevsr @paulbalandan 

> Using preloading requires one dedicated PHP. Normally, web servers are configured to use one PHP, so one app requires a dedicated web server.
>
> If you want to use preloading for multiple apps on one web server, configure it to use multiple PHP-FPMs, and one virtual host uses one PHP-FPM.

Reply:
In the 2nd paragraph, do you mean to say "configure your server to use virtual hosts with multiple PHP-FPMs, with each virtual host using one PHP-FPM". I have not used PHP-FPM before so I'm not sure if what I'm saying is correct.

Reply:
Yes!

Like this:
```
<VirtualHost *:80>
    ServerName php81.example.com
    DocumentRoot /var/www/php81
    <Directory /var/www/php81>
        Options -Indexes +FollowSymLinks +MultiViews
        AllowOverride All
        Require all granted
    </Directory>
    <FilesMatch \.php$>
        # Apache 2.4.10+ can proxy to unix socket
        SetHandler "proxy:unix:/var/run/php/php8.1-fpm.sock|fcgi://localhost"
    </FilesMatch>
</VirtualHost>


<VirtualHost *:80>
    ServerName php74.example.com
    DocumentRoot /var/www/php74
    <Directory /var/www/php74>
        Options -Indexes +FollowSymLinks +MultiViews
        AllowOverride All
        Require all granted
    </Directory>
    <FilesMatch \.php$>
        SetHandler "proxy:unix:/var/run/php/php7.4-fpm.sock|fcgi://localhost"
    </FilesMatch>
</VirtualHost>
```
https://tecadmin.net/install-multiple-php-version-apache-ubuntu/

Reply:
@kenjis @paulbalandan Handler PHP not just PHP-FPM. I used LSAPI too in my webserver openlitespeed

Ref : https://www.hostdime.com/kb/hd/miscellaneous/what-are-php-handlers-and-why-do-they-matter#WhatarePHPHandlersandWhyDoTheyMatter?-FCGI/FastCGI

Reply:
Then, how about this?

> Using preloading requires one dedicated PHP handler. Normally, web servers are configured to use one PHP handler, so one app requires a dedicated web server.
>
> If you want to use preloading for multiple apps on one web server, configure your server to use virtual hosts with multiple PHP handlers like multiple PHP-FPMs, with each virtual host using one PHP handler.

@ddevsr LSAPI is a Server API, so it is not good word for a handler.

> The LSAPI is a highly optimized API that allows communication between LiteSpeed and third party web engines. 
https://www.php.net/manual/en/install.unix.litespeed.php#install.unix.litespeed

If you want to show the settings in LiteSpeed, add the links you put.

Reply:
@kenjis I am OK with that

> Using preloading requires one dedicated PHP handler. Normally, web servers are configured to use one PHP handler, so one app requires a dedicated web server.
> 
> If you want to use preloading for multiple apps on one web server, configure your server to use virtual hosts with multiple PHP handlers like multiple PHP-FPMs, with each virtual host using one PHP handler.